### DISCUSSION TITLE
No Audio from CarPlay message reading

### DISCUSSION THREAD

**Discussion by gernb** [tesla-android/issue-tracker#154] [Q&A] [answered]
Hi. First, many thanks to Mike for this awesome, open source project!

I recently installed 2022.44.2 on a single RPi4, connected my CarLinkIt dongle, and configured everything as described on the install guide for CarPlay. Things generally work pretty well and I am quite pleased with how much everything “just works.” However, I have a couple of issues that I wonder if others have encountered and know the solution to.
1. When I receive a message and tap the banner (in CarPlay) to listen to the received message, the UI shows it is reading the message, but I cannot hear anything. Other audio (music, navigation, etc) works fine.
2. Every time my phone connects to CarPlay I have change the audio route (on my phone) from the CarPlay audio route to the Tesla bluetooth connection. Is that normal? Is there a way to configure things so that audio routes to the car automatically?

Thanks!
-peter

**Comment by adamChrystie** [comment]
@gernb Ensure in autokit advanced settings that you set audio destination to be Bluetooth and not box as mentioned in the tesla-android instructions. Also in autokit settings try setting microphone source to be phone instead of car. In Tesla’s Bluetooth settings I mark my phone as priority device. 

I occasionally have the issue you describe where there is no audio from nav & Siri when reading messages but I hear music & podcasts. I am not sure what causes this situation but seems to happen usually after I push refresh on tesla browser to correct the occasional situation where the browser stops updating the display as if it Lost connection to the pi 4. I am usually able to solve this by going to tesla Bluetooth audio  source gui and push pause and then play again. Maybe optionally afterwards followed by pausing and playing music from CarPlay gui.

I have not yet had the issue where I need to manually switch my call audio route from CarPlay to phone bluetooth. I do not use phone often but so far when I receive/make calls, it works great & better than my old hansshow console. I have iPhone 13pro max and iOS 16.1.1  . My iPhones call routing option is set to automatic. Perhaps you have a different iPhone which behaves differently. When I had hansshow 9” linux console and older iOS & same phone, I ran in to audio routing issue you describe. Setting phone’s call routing option to speaker or headset seemed to help.

**Reply by gernb** [reply]
Thanks for the reply and the info, Adam.
I guess I should have been more clear in my original post. I did ensure that the audio destination is set to BT and not Box. I have also tried playing around with a bunch of other settings and have had no success so far.

**Comment by gernb** [comment, marked as answer]
Problem resolved. Apparently my Carlinkit dongle was too old (or, at least the firmware on it was). After purchasing a newer dongle all is working as expected, including message reading and default audio routing. I’m not sure why the Autokit app didn’t update the firmware on my older dongle, but the new one works great.
